File: code-coaches/medium/youtube-link-finder/YouTubeLinkFinder.swift
import Foundation

func extractVideoId(from url: String) throws -> String {
    if url.contains("youtube.com/watch") {
        let parts = url.components(separatedBy: "v=")
        if parts.count == 2 {
            return parts[1]
        }
    } else if url.contains("youtu.be/") {
        let parts = url.components(separatedBy: "/")
        if let lastPart = parts.last {
            return lastPart
        }
    }
    throw NSError(domain: "Invalid YouTube video URL", code: 0)
}

func main() {
    if let url = readLine() {
        do {
            let videoId = try extractVideoId(from: url)
            print(videoId)
        } catch let error as NSError {
            print("Error: \(error.localizedDescription)")
        }
    }
}

main()

File: code-coaches/medium/youtube-link-finder/youtube_link_finder.c
#include <stdio.h>
#include <string.h>

char* extract_video_id(char* url) {
    char* id = NULL;
    char* pos = NULL;

    if ((pos = strstr(url, "v=")) != NULL) {
        id = pos + 2; // add 2 to skip "v="
    } else if ((pos = strstr(url, "be/")) != NULL) {
        id = pos + 3; // add 3 to skip "be/"
    }

    return id;
}

int main() {
    char url[256];
    fgets(url, 256, stdin);

    char* id = extract_video_id(url);
    if (id == NULL) {
        printf("Invalid YouTube URL.\n");
    } else {
        printf("%s", id);
    }

    return 0;
}
